superdesk/types/archive.py

```diff
@@ -0,0 +1,453 @@
+class MetadataResource(ResourceModel):
+    guid: str
+    uri: str
+    unique_id: int
+    unique_name: str
+    version: int
+    ingest_id: str
+    ingest_version: str
+    family_id: str
+    related_to: str
+    original_creator: Annotated[str | None, validate_data_relation_async("users")] = None
+    version_creator: Annotated[str | None, validate_data_relation_async("users")] = None
+    firstcreated: datetime
+    versioncreated: datetime
+    firstpublished: datetime | None = None
+    ingest_provider: Annotated[str | None, validate_data_relation_async("ingest_providers")] = None
+    source: str
+    original_source: str
+    ingest_provider_sequence: str
+    usageterms: str | None = None
+    copyrightnotice: str | None = None
+    copyrightholder: str | None = None
+    anpa_category: list[Subject] | None = None
+    subject: list[Subject]
+    genre: list[Genre] | None = None
+    company_codes: list[CompanyCodes] | None = None
+    item_type: ContentTypes = Field(ContentTypes.TEXT, alias="type")
+    package_type: str | None = Field(None, deprecated=True)
+    language: str | None = None
+    abstract: str | None = None
+    headline: str | None = None
+    slugline: Slugline
+    anpa_take_key: str | None = None
+    correction_sequence: int | None = None
+    rewrite_sequence: int | None = None
+    # FIXME: duplicate of ItemSchema
+    rewrite_of: str | None = None
+    # FIXME: duplicate of ItemSchema
+    rewritten_by: str | None = None
+    # FIXME: duplicate of ItemSchema
+    sequence: int | None = None
+    keywords: list[str]
+    word_count: int
+    priority: int | None = None
+    urgency: int | None = None
+    profile: str | None = None
+    state: ContentStates
+    revert_state: ContentStates | None = None
+    pubstatus: PubStatuses | None = None
+    signal: list[Subject]
+    byline: str | None = None
+    ednote: str | None = None
+    authors: list[Author] | None = None
+    description_text: str | None = None
+    archive_description: str | None = None
+    groups: list[GroupRef] | None = None
+    deleted_groups: list[str] | None = None
+    body_html: str | None = None
+    body_text: str | None = None
+    dateline: Dateline | None = None
+    expiry: datetime
+    # FIXME: type file
+    media: str | None = None
+    mimetype: str
+    poi: POI
+    renditions: dict[str, dict]
+    filemeta: dict[str, dict]
+    filemeta_json: str
+    media_file: str
+    contents: list
+    associations: Associations
+    refs: list[Ref]
+    alt_text: str | None = None
+    # FIXME: Place is slightly different ("location" is not a list here)
+    place: list[Place] | None = None
+    event: EntityMetadata | None = None
+    person: EntityMetadata | None = None
+    object: EntityMetadata | None = None
+    organisation: EntityMetadata | None = None
+    creditline: str
+    linked_in_packages: list[LinkedInPackage] | None = None
+    highlight: Annotated[str | None, validate_data_relation_async("highlights")] = None
```

Comment:
Should these use [fields.ObjectId](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/fields.py#L150), so serialisation works as expected.